users/arxiv/users/legacy/authenticate.py

```diff
@@ -143,9 +146,28 @@ def _authenticate_password(username_or_email: str, password: str) -> PassData:
     ------
     :class:`AuthenticationFailed`
         Raised if the user does not exist or the password is incorrect.
+    :class:`RuntimeError`
+        Raised when other problems arise.
 
     """
     logger.debug(f'Authenticate with password, user: {username_or_email}')
+
+    if not password:
+        raise RuntimeError('Passed empty password')
+    if not isinstance(password, str):
+        raise RuntimeError('Passed non-str password: {type(password)}')
+    if not is_ascii(password):
+        raise RuntimeError('Password non-ascii password')
+
+    if not username_or_email:
+        raise RuntimeError('Passed empty username_or_email')
+    if not isinstance(password, str):
+        raise RuntimeError('Passed non-str username_or_email: {type(username_or_email)}')
+    if len(username_or_email) > 255:
+        raise RuntimeError(f'Passed username_or_email too long: len {len(username_or_email)}')
```

Comment:
New checks look good. We're stuck with 90s-ish ASCII. I'm curious whether a super long username or email could be indication of 'abuse'. We should monitor the logs after this is deployed. Where these new checks all derived from reviewing the bad requests?

Reply:
The new checks are based on the database fields and the constraints on the HTMLForms in arXivSubmit for creating a new user.